### PR TITLE
MediaPlayer.WMS now changes stream volume

### LINKED ISSUE #2780: MediaPlayer.Volume affects all sounds
It seems like whenever I try to use MediaPlayer.Volume to change the volume of a background song it affects all sounds (sounds playing via SoundEffect). This is on Windows 7 using an unstable build I just downloaded from TeamCity. The stable 3.2 build does this too. My code works fine using XNA.

### PR DESCRIPTION
**Pull request by KonajuGames**
The previous iteration changed the application volume controller which affected all sound output including sound effects. It now uses the stream volume controller so it only affects the streaming media and sound effects remain unaffected.

Thanks to MarkoBL for the fix.

Fixes #2780

### CODE REVIEW

File: MonoGame.Framework/Media/MediaPlayer.WMS.cs [hunk 1]
```diff
@@ -124,29 +137,39 @@ private static void PlatformPlaySong(Song song)
...
-            _volumeController.MasterVolume = _volume;
+            // Get the volume interface.
+            IntPtr volumeObj;
+            try
+            {
+                MediaFactory.GetService(_session, MRPolicyVolumeService, AudioStreamVolumeGuid, out volumeObj);
+            }
+            catch
+            {
+                MediaFactory.GetService(_session, MRPolicyVolumeService, AudioStreamVolumeGuid, out volumeObj);
```
Comment (tomspilman) [inline MonoGame.Framework/Media/MediaPlayer.WMS.cs]: Is this happening here again?  Where we need to get the service twice as it sometimes fails?

Comment (KonajuGames) [inline MonoGame.Framework/Media/MediaPlayer.WMS.cs]: It could be. @MarkoBL, is that correct?

Comment (tomspilman) [inline MonoGame.Framework/Media/MediaPlayer.WMS.cs]: @MarkoBL ?

**Comment by MarkoBL** [comment]
I can't follow you exactly, what you're talking about. I'm not an XAudio expert, I only changed the ```SimpleAudioVolume``` to ```AudioStreamVolume```to fix the inital problem.

**Comment by tomspilman** [comment]
@MarkoBL - The code you posted here:

 https://github.com/mono/MonoGame/issues/2780#issuecomment-53998018

Included this...
```
            try
            {
                MediaFactory.GetService(_session, MRPolicyVolumeService, AudioStreamVolumeGuid, out volumeObj);
            }
            catch
            {
                MediaFactory.GetService(_session, MRPolicyVolumeService, AudioStreamVolumeGuid, out volumeObj);
            }  
```
Did you add this try/catch and retry code because you saw this fail?  Or was this already in the code?

**Comment by MarkoBL** [comment]
This was already in the code.

**Comment by tomspilman** [comment]
@MarkoBL - Thanks for clearing that up!

File: MonoGame.Framework/Media/MediaPlayer.WMS.cs (hunk 1 above)
Comment (tomspilman) [inline MonoGame.Framework/Media/MediaPlayer.WMS.cs]: So the code we already have that is similar to this yet improved:

https://github.com/mono/MonoGame/blob/develop/MonoGame.Framework/Media/VideoPlayer.WMS.cs#L126

Just a different GUID, but same sort of issue.

@KonajuGames - Should we maybe call this from here too?

**Comment by Spectrallic** [comment]
Is this already merged? If not, any idea when it will be?

File: MonoGame.Framework/Media/MediaPlayer.WMS.cs (hunk 1 above)
Comment (tomspilman) [inline MonoGame.Framework/Media/MediaPlayer.WMS.cs]: @KonajuGames - Bump... for when you get back home. :)

**Comment by tomspilman** [comment]
@Spectrallic - Soon... waiting on @KonajuGames for something.

File: MonoGame.Framework/Media/MediaPlayer.WMS.cs (hunk 1 above)
Comment (tomspilman) [inline MonoGame.Framework/Media/MediaPlayer.WMS.cs]: @KonajuGames - Bump!

Comment (KonajuGames) [inline MonoGame.Framework/Media/MediaPlayer.WMS.cs]: I can add that in.

**Comment by KonajuGames** [comment]
MediaPlayer and VideoPlayer are now consistent and correct when controlling volumes.

**Comment by tomspilman** [comment]
Great... merging!

**Pull request merged**